### PR TITLE
Fix Hydraulics stops working when using setVehicleHandling with player inside vehicle

### LINKED ISSUE #1121: Hydraulics stops working when using setVehicleHandling with player inside vehicle
**Describe the bug**
When you set maxVelocity with setVehicleHandling function and you're inside vehicle, you can't use it. Hydraulics is still mounted but vehicle behave like it has no hydraulics, you can honk instead of lifting a car.

**To reproduce**
1. create vehicle with upgrade 1087
2. try to use hydraulics [you can]
3. run vehicle = getPedOccupiedVehicle(getPlayerFromName("MrDadosz")); setVehicleHandling(vehicle, "maxVelocity", 100)
4. try to use hydraulics [you can't]

**Version**
v1.5.7-release-20288

**Additional context**
You can fix it with script:
```lua
if getVehicleUpgradeOnSlot(vehicle, 9) == 1087 then
	removeVehicleUpgrade(vehicle, 1087)
	addVehicleUpgrade(vehicle, 1087)
end
```
But it shouldn't happen.

### PR DESCRIPTION
**Pull request by FileEX**
Fixed #1121

### CODE REVIEW

File: Client/game_sa/CVehicleSA.cpp [hunk 1]
```diff
@@ -1322,49 +1333,41 @@ void CVehicleSA::RecalculateHandling()
...
-        {
-            // Unset the flag
-            uiHandlingFlags &= ~HANDLING_Hydraulics_Flag;
-            m_pHandlingData->SetHandlingFlags(uiHandlingFlags);
+            hydralicsInstalled = true;
         }
     }
+
+    // If hydraulics isn't installed we need unset the flag
+    if (!hydralicsInstalled && uiHandlingFlags & HANDLING_Hydraulics_Flag)
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Please, use brackets to emphasize the order of operations.

File: Client/game_sa/CVehicleSA.cpp [hunk 2]
```diff
@@ -1322,49 +1333,41 @@ void CVehicleSA::RecalculateHandling()
...
+        uiHandlingFlags &= ~HANDLING_Hydraulics_Flag;
+
+    // If NOS isn't installed we need unset the flag
+    if (!nitroInstalled && uiHandlingFlags & HANDLING_NOS_Flag)
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Please, use brackets to emphasize the order of operations.

File: Client/game_sa/CVehicleSA.cpp [hunk 3]
```diff
@@ -1322,49 +1333,41 @@ void CVehicleSA::RecalculateHandling()
...
         }
-    }
-    // Hydraulics Flag fixing
-    if (pInt->m_upgrades[1] && pInt->m_upgrades[1] == 1087)
-    {
-        // Flag not enabled?
-        if (uiHandlingFlags | HANDLING_Hydraulics_Flag)
+
+        // If hydraulics is installed we need set the flag
+        if (upgradeID == 1087 && uiHandlingFlags | HANDLING_Hydraulics_Flag)
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Please, use brackets to emphasize the order of operations.

File: Client/game_sa/CVehicleSA.cpp [hunk 4]
```diff
@@ -1322,49 +1333,41 @@ void CVehicleSA::RecalculateHandling()
...
+    for (auto& upgradeID : pInt->m_upgrades)
     {
-        // Flag not enabled?
-        if (uiHandlingFlags | HANDLING_NOS_Flag)
+        // Empty upgrades value is -1
+        if (upgradeID < 0)
+            continue;
+
+        // If NOS is installed we need set the flag
+        if ((upgradeID >= 1008 && upgradeID <= 1010) && uiHandlingFlags | HANDLING_NOS_Flag)
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Please, use brackets to emphasize the order of operations.

File: Client/game_sa/CVehicleSA.cpp [hunk 5]
```diff
@@ -684,6 +684,17 @@ void CVehicleSA::RemoveVehicleUpgrade(DWORD dwModelID)
         push    dwModelID
         call    dwFunc
     }
+
+    // GTA SA only does this when CVehicle::ClearVehicleUpgradeFlags returns false.
+    // In the case of hydraulics and nitro, this function does not return false and the upgrade is never removed from the array
+    for (auto& upgrade : GetVehicleInterface()->m_upgrades)
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Please, do not use `auto` instead of short typenames.

File: Client/game_sa/CVehicleSA.cpp [hunk 6]
```diff
@@ -1322,49 +1333,41 @@ void CVehicleSA::RecalculateHandling()
     // Put it in our interface
     CVehicleSAInterface* pInt = GetVehicleInterface();
     unsigned int         uiHandlingFlags = m_pHandlingData->GetInterface()->uiHandlingFlags;
-    // user error correction - NOS_INST = NOS Installed t/f
-    // if nos is installed we need the flag set
-    if (pInt->m_upgrades[0] && pInt->m_upgrades[0] >= 1008 && pInt->m_upgrades[0] <= 1010)
+    bool                 hydralicsInstalled, nitroInstalled;
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Do not foget to initilize these variables.

File: Client/game_sa/CVehicleSA.cpp [hunk 7]
```diff
@@ -1322,49 +1333,41 @@ void CVehicleSA::RecalculateHandling()
...
+
+    // We check whether the user has not set incorrect flags via handlingFlags in the case of nitro and hydraulics
+    // If this happened, we need to correct it
+    for (auto& upgradeID : pInt->m_upgrades)
```
Comment (tederis) [inline Client/game_sa/CVehicleSA.cpp]: Please, do not use `auto` instead of short typenames.

**Review by tederis — CHANGES_REQUESTED** [review]

File: Client/game_sa/CVehicleSA.cpp (hunk 7 above)
Comment (TracerDS) [reply Client/game_sa/CVehicleSA.cpp]: If possible, also use `const`

File: Client/game_sa/CVehicleSA.cpp (hunk 5 above)
Comment (TracerDS) [reply Client/game_sa/CVehicleSA.cpp]: If possible, also use `const`

Comment (FileEX) [reply Client/game_sa/CVehicleSA.cpp]: This cannot be const because the value is changed to -1

Comment (TracerDS) [reply Client/game_sa/CVehicleSA.cpp]: Didnt see that, github mobile truncates the file to only changed lines

File: Client/game_sa/CVehicleSA.cpp [hunk 8]
```diff
@@ -687,7 +687,7 @@ void CVehicleSA::RemoveVehicleUpgrade(DWORD dwModelID)
 
     // GTA SA only does this when CVehicle::ClearVehicleUpgradeFlags returns false.
     // In the case of hydraulics and nitro, this function does not return false and the upgrade is never removed from the array
-    for (auto& upgrade : GetVehicleInterface()->m_upgrades)
+    for (std::int16_t upgrade : GetVehicleInterface()->m_upgrades)
```
Comment (TracerDS) [inline Client/game_sa/CVehicleSA.cpp]: why did you remove the reference?

File: Client/game_sa/CVehicleSA.cpp [hunk 9]
```diff
@@ -1333,37 +1333,37 @@ void CVehicleSA::RecalculateHandling()
     // Put it in our interface
     CVehicleSAInterface* pInt = GetVehicleInterface();
     unsigned int         uiHandlingFlags = m_pHandlingData->GetInterface()->uiHandlingFlags;
-    bool                 hydralicsInstalled, nitroInstalled;
+    bool                 hydralicsInstalled = false, nitroInstalled = false;
 
     // We check whether the user has not set incorrect flags via handlingFlags in the case of nitro and hydraulics
     // If this happened, we need to correct it
-    for (auto& upgradeID : pInt->m_upgrades)
+    for (const std::int16_t upgradeID : pInt->m_upgrades)
```
Comment (TracerDS) [inline Client/game_sa/CVehicleSA.cpp]: why did you remove the reference?

File: Client/game_sa/CVehicleSA.cpp [hunk 10]
```diff
@@ -1333,37 +1333,37 @@ void CVehicleSA::RecalculateHandling()
...
     {
         // Empty upgrades value is -1
         if (upgradeID < 0)
             continue;
 
         // If NOS is installed we need set the flag
-        if ((upgradeID >= 1008 && upgradeID <= 1010) && uiHandlingFlags | HANDLING_NOS_Flag)
+        if ((upgradeID >= 1008 && upgradeID <= 1010) && (uiHandlingFlags | HANDLING_NOS_Flag))
```
Comment (TracerDS) [inline Client/game_sa/CVehicleSA.cpp]: what is `uiHandlingFlags | HANDLING_NOS_Flag` supposed to do in your opinion?

File: Client/game_sa/CVehicleSA.cpp [hunk 11]
```diff
@@ -1333,37 +1333,37 @@ void CVehicleSA::RecalculateHandling()
...
         {
             uiHandlingFlags |= HANDLING_NOS_Flag;
             nitroInstalled = true;
         }
 
         // If hydraulics is installed we need set the flag
-        if (upgradeID == 1087 && uiHandlingFlags | HANDLING_Hydraulics_Flag)
+        if ((upgradeID == 1087) && (uiHandlingFlags | HANDLING_Hydraulics_Flag))
```
Comment (TracerDS) [inline Client/game_sa/CVehicleSA.cpp]: above

**Review by TracerDS — CHANGES_REQUESTED** [review]

File: Client/game_sa/CVehicleSA.cpp (hunk 10 above)
Comment (FileEX) [reply Client/game_sa/CVehicleSA.cpp]: I was relying on the default code and didn't even notice this mistake, ty

**Comment by Dutchman101** [comment]
Thanks, this is certainly a desired fix.
It passed 2 code reviews, and feedback has been addressed.

**Pull request merged**